pkg/archive/archive_unix.go

```diff
@@ -109,7 +108,7 @@ func handleTarTypeBlockCharFifo(hdr *tar.Header, path string) error {
 		mode |= unix.S_IFIFO
 	}
 
-	return system.Mknod(path, mode, int(system.Mkdev(hdr.Devmajor, hdr.Devminor)))
+	return unix.Mknod(path, mode, int(unix.Mkdev(uint32(hdr.Devmajor), uint32(hdr.Devminor))))
```

Comment:
This needs differentiation, because the new implementation doesn't support (Free)BSD; https://github.com/moby/buildkit/pull/5602

```
49.21 # github.com/docker/docker/pkg/archive
49.21 vendor/github.com/docker/docker/pkg/archive/archive_unix.go:111:32: cannot use int(unix.Mkdev(uint32(hdr.Devmajor), uint32(hdr.Devminor))) (value of type int) as uint64 value in argument to unix.Mknod
```

Reply:
Yes see also https://github.com/tonistiigi/fsutil/blob/bec5f0e1255330c04d8c05315b5a3cd3fa3e1590/diskwriter_freebsd.go#L12